common/types/activation.go

```diff
@@ -317,6 +296,54 @@ func (atx *ActivationTx) GetShortPoetProofRef() []byte {
 	return ref[:util.Min(5, len(ref))]
 }
 
+// ShortString returns the first 5 characters of the ID, for logging purposes.
+func (atx *ActivationTx) ShortString() string {
+	return atx.ID().ShortString()
+}
+
+// ID returns the ATX's ID.
+func (atx *ActivationTx) ID() ATXID {
+	if atx.id == nil {
+		panic("id field must be set")
+	}
+	return *atx.id
+}
+
+// NodeID returns the ATX's Node ID.
+func (atx *ActivationTx) NodeID() NodeID {
+	if atx.nodeID == nil {
+		panic("nodeID field must be set")
+	}
+	return *atx.nodeID
+}
+
+// TargetEpoch returns the target epoch of the activation transaction. This is the epoch in which the miner is eligible
+// to participate thanks to the ATX.
+func (atx *ActivationTx) TargetEpoch() EpochID {
+	return atx.PubLayerID.GetEpoch() + 1
+}
+
+// SetID sets the ATXID in this ATX's cache.
+func (atx *ActivationTx) SetID(id *ATXID) {
+	atx.id = id
+}
+
+// SetNodeID sets the Node ID in the ATX's cache.
+func (atx *ActivationTx) SetNodeID(nodeID *NodeID) {
+	atx.nodeID = nodeID
+}
+
+// Verify an ATX for a given base TickHeight and TickCount.
+func (atx *ActivationTx) Verify(baseTickHeight, tickCount uint64) *VerifiedActivationTx {
```

Comment:
This method's comment and name imply that this function performs **verification**. But it only constructs a "verified" type assuming the caller has already verified it somehow. It's misleading. Perhaps rename the method to something like `ToVerified`.